modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -116,10 +115,10 @@ public static function validateForm(array &$element, FormStateInterface $form_st
     if (empty($coupon_code)) {
       return;
     }
+    $entity_type_manager = \Drupal::entityTypeManager();
     $code_path = implode('][', $coupon_parents);
     /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
-    $order = $form_state->get('order');
-    $entity_type_manager = \Drupal::entityTypeManager();
+    $order = $entity_type_manager->getStorage('commerce_order')->load($element['#order']->id());
```

Comment:
@bojanz do you want me to bust the order storage into its own var, or is this fine.